src/MoBi.Presentation/DTO/NeighborhoodBuilderDTO.cs

```diff
@@ -1,15 +1,56 @@
+using System.Collections.Generic;
+using System.Linq;
 using OSPSuite.Assets;
+using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
 
 namespace MoBi.Presentation.DTO
 {
    public class NeighborhoodBuilderDTO : ContainerDTO
    {
-      public string FirstNeighborPath { get; set; }
-      public string SecondNeighborPath { get; set; }
-      public NeighborhoodBuilderDTO(NeighborhoodBuilder neighborhoodBuilder) : base(neighborhoodBuilder)
+      private Cache<string, List<string>> _connections;
+      public NeighborhoodObjectPathDTO FirstNeighborDTO { get; }
+      public NeighborhoodObjectPathDTO SecondNeighborDTO { get; }
+      public string FirstNeighborPath => FirstNeighborDTO.Path;
+      public string SecondNeighborPath => SecondNeighborDTO.Path;
+
+      public NeighborhoodBuilderDTO(NeighborhoodBuilder neighborhoodBuilder, IReadOnlyList<NeighborhoodBuilder> existingNeighborhoods) : base(neighborhoodBuilder)
       {
          Icon = ApplicationIcons.Neighborhood;
+         FirstNeighborDTO = new NeighborhoodObjectPathDTO(this, () => SecondNeighborDTO) { Path = neighborhoodBuilder.FirstNeighborPath?.ToPathString() };
+         SecondNeighborDTO = new NeighborhoodObjectPathDTO(this, () => FirstNeighborDTO) { Path = neighborhoodBuilder.SecondNeighborPath?.ToPathString() };
+         addExistingNeighborhoods(existingNeighborhoods);
+      }
+
+      private void addExistingNeighborhoods(IReadOnlyList<NeighborhoodBuilder> existingNeighborhoods)
+      {
+         var uniqueNeighbors = existingNeighborhoods.SelectMany(x => new[] { x.FirstNeighborPath, x.SecondNeighborPath }).Distinct().Where(x => x != null);
+         _connections = new Cache<string, List<string>>();
+
+         uniqueNeighbors.Each(x =>
+         {
+            existingNeighborhoods.Where(y => Equals(y.FirstNeighborPath, x)).Each(y => addConnection(_connections, x, y.SecondNeighborPath));
+            existingNeighborhoods.Where(y => Equals(y.SecondNeighborPath, x)).Each(y => addConnection(_connections, x, y.FirstNeighborPath));
+         });
+      }
```

Comment:
don't understand the workflow. Why not just 
`existingNeighborhoods.Each(nbh => addConnection(_connections, nbh.FirstNeighborPath, nbh.SecondNeighborPath)`

And `addConnection()` should check that both paths are not null before adding the connection